app/services/idv/steps/inherited_proofing/verify_wait_step_show.rb

```diff
@@ -0,0 +1,75 @@
+module Idv
+  module Steps
+    module InheritedProofing
+      class VerifyWaitStepShow < VerifyBaseStep
+        include UserPiiManagable
+        delegate :controller, :idv_session, to: :@flow
+
+        STEP_INDICATOR_STEP = :getting_started
+
+        def call
+          poll_with_meta_refresh(IdentityConfig.store.poll_rate_for_verify_in_seconds)
+
+          process_async_state(async_state)
+        end
+
+        private
+
+        def process_async_state(current_async_state)
+          if current_async_state.none?
+            mark_step_incomplete(:agreement)
+          elsif current_async_state.in_progress?
+            nil
+          elsif current_async_state.missing?
+            flash[:error] = I18n.t('idv.failure.timeout')
+            # Need to add path to error pages once they exist
+            # LG-7257
+            # This method overrides VerifyBaseStep#process_async_state:
+            # See the VerifyBaseStep#process_async_state "elsif current_async_state.missing?"
+            # logic as to what is typically needed/performed when hitting this logic path.
+          elsif current_async_state.done?
+            async_state_done(current_async_state)
+          end
+        end
+
+        def async_state
+          return ProofingSessionAsyncResult.none if dcs_uuid.nil?
+          return ProofingSessionAsyncResult.missing if document_capture_session.nil?
+          return ProofingSessionAsyncResult.missing if api_job_result.nil?
+
+          api_job_result
+        end
+
+        def async_state_done(_current_async_state)
+          service_provider = controller.inherited_proofing_service_provider_id
+          form = Idv::InheritedProofing::ServiceProviderFormFactory.form(
+            service_provider: service_provider, payload_hash: api_job_result[:result],
+          )
+          form_response = form.submit
+
+          if form_response.success?
+            flow_session[:pii_from_user] = form.user_pii
```

Comment:
Not so sure we want to simply assign here. The pattern I've seen is that we merge - I believe this is because if there are (ever) any steps inserted _before_ this step that gathers pii, it doesn't get wiped. Either way, it wouldn't hurt. Take a look at what `UserPiiManageable#inherited_proofing_save_session!` does - you want to do the same.

I also believe we want to make sure we perform the same logic in `UserPiiManageable#inherited_proofing_skip_steps!` << this we may want to set after they verify their info perhaps on the next step? @jmhooper question. I think it's more appropriate there.

Reply:
May want to refactor a little and use `UserPiiManagable' after all.

Reply:
You're totally right and that got mangled and lost during a refactor at some point. Thanks for catching that! (And all the other catches. I'm so tired.... 😪 )